### ISSUE TITLE
Spec restricts uses of const enum from type annotations

### ISSUE THREAD

**Issue by yuit** [Microsoft/TypeScript#2786]
According to section [9.4] (https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#94-constant-enum-declarations)

> unlike regular enum declarations, constant enum declarations are completely erased in the emitted JavaScript code. For this reason, it is an error to reference a constant enum object in any other context than a property access that selects one of the enum's members.

Although, the compiler will erase any type annotation in emitted JavaScript code, it will be more consistent with the spec if we give an error

For example:
```ts
const enum E {}
var e: E;  // no error given
```

**Comment by DanielRosenwasser** [comment]
This sounds more like a spec bug than a compiler bug.

**Comment by ahejlsberg** [comment]
The spec states that it is an error to reference a constant enum ***object*** in any other context than than a property access that selects one of the enum's members. This doesn't mean that it is an error to reference a constant enum ***type***, which is what your example does. I think [section 9.1](https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#91-enum-declarations) in the spec is already pretty clear that an enum declaration introduces an ***enum type*** and an ***enum object***.
